src/pybind/mgr/dashboard/frontend/src/app/shared/components/grafana/grafana.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Input } from '@angular/core';
+import { OnChanges } from '@angular/core';
+import { OnInit } from '@angular/core';
+import { Output } from '@angular/core';
+
+import { DomSanitizer } from '@angular/platform-browser';
+import { SafeUrl } from '@angular/platform-browser';
+
+import { GrafanaService } from '../../../shared/api/grafana.service';
+
+import * as grafanaConst from '../../enum/grafana-template.enum';
+
+@Component({
+  selector: 'cd-grafana',
+  templateUrl: './grafana.component.html',
+  styleUrls: ['./grafana.component.scss']
+})
+export class GrafanaComponent implements OnInit, OnChanges {
+  grafanaSrc: SafeUrl;
+  url: string;
+  protocol: string;
+  host: string;
+  port: number;
+  baseUrl: any;
+  dashboardElement: string;
+  elementId: string;
+  dashboardVar: string;
+  grafanaExist = false;
+  dashboardPath: string;
+
+  @Input()
+  grafanaPath: string;
+  @Input()
+  grafanaStyle: string;
+
+  @Output()
+  fetchData = new EventEmitter();
+  panelStyle: any;
+
+  constructor(private sanitizer: DomSanitizer, private grafanaService: GrafanaService) {}
+
+  ngOnInit() {
+    this.grafanaService.getProtocol().subscribe((protocolData: any) => {
+      this.protocol = protocolData.value;
+      this.grafanaService.getHost().subscribe((hostData: any) => {
+        this.host = hostData.value;
+        this.grafanaService.getPort().subscribe((portData: any) => {
+          this.port = portData.value;
+          this.getFrame();
+        });
+      });
+    });
+  }
+
+  getFrame() {
+    this.baseUrl = this.protocol + '://' + this.host + ':' + String(this.port) + '/d/';
+    this.grafanaExist = true;
+    this.dashboardPath = this.grafanaPath;
+    this.url = this.baseUrl + this.dashboardPath + '&refresh=2s';
+    this.panelStyle = grafanaConst[this.grafanaStyle];
+    this.grafanaSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.url);
+  }
+
+  ngOnChanges(changes) {
+    this.ngOnInit();
```

Comment:
Since I don't think grafana settings are going to be modified that often, I would suggest to stop calling `ngOnInit` and call directly `getFrame`.

Reply:
ack